packages/http-client-csharp/generator/Microsoft.Generator.CSharp/test/Providers/ModelProviders/ModelCustomizationTests.cs

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.Generator.CSharp.Input;
+using Microsoft.Generator.CSharp.Providers;
+using Microsoft.Generator.CSharp.Tests.Common;
+using NUnit.Framework;
+
+namespace Microsoft.Generator.CSharp.Tests.Providers // the namespace here is crucial to get correct test data file.
+{
+    public class ModelCustomizationTests
+    {
+        // Validates that the property body's setter is correctly set based on the property type
+        [TestCase]
+        public void TestCustomization_CanChangeModelName()
+        {
+            // we are borrowing this GetExpectedFromFile method to read the content of the corresponding asset file, we are not using it as expected here.
+            SyntaxTree syntaxTree = CSharpSyntaxTree.ParseText(Helpers.GetExpectedFromFile());
```

Comment:
this compilation looks boilerplate can we make it a helper somewhere in LoadMockPlugin or MockHelpers?

Reply:
I think this should resolve my concern [here](https://github.com/microsoft/typespec/pull/4285#discussion_r1735584868) as well

Reply:
Done!